### PR TITLE
Run `flake8-noqa` in CI

### PR DESCRIPTION
**Pull request by AlexWaygood**
[flake8-noqa](https://github.com/plinss/flake8-noqa) is a plugin for validating `noqa` comments. Mypy recently added it to their CI, and I think it would be good for us to add it as well.

Unfortunately, we cannot enable `NQA102`, probably the most useful `flake8-noqa` check, which detects unused `noqa` comments. This is because we quite often have unused `noqa` comments in typeshed for unreleased flake8-pyi checks.

Nonetheless, flake8-noqa has a number of other useful checks, so I think it would be a worthy addition to our CI anyway.

### CODE REVIEW

**Review by srittau — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -102,7 +102,8 @@ black .
 ...Or install the pre-commit hooks: please refer to the
 [pre-commit](https://pre-commit.com/) documentation.
 
-Our code is also linted using `flake8`, with plugins `flake8-pyi` and `flake8-bugbear`. As with our other checks, running
+Our code is also linted using `flake8`, with plugins `flake8-pyi`
```
Comment (AlexWaygood) [inline CONTRIBUTING.md:105]:
```suggestion
Our code is also linted using `flake8`, with plugins `flake8-pyi`,
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -102,7 +102,8 @@ black .
 ...Or install the pre-commit hooks: please refer to the
 [pre-commit](https://pre-commit.com/) documentation.
 
-Our code is also linted using `flake8`, with plugins `flake8-pyi` and `flake8-bugbear`. As with our other checks, running
+Our code is also linted using `flake8`, with plugins `flake8-pyi`,
+`flake8-bugbear`, and flake8-noqa. As with our other checks, running
```
Comment (AlexWaygood) [inline CONTRIBUTING.md:106]:
```suggestion
`flake8-bugbear`, and `flake8-noqa`. As with our other checks, running
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -102,7 +102,8 @@ black .
 ...Or install the pre-commit hooks: please refer to the
 [pre-commit](https://pre-commit.com/) documentation.
 
-Our code is also linted using `flake8`, with plugins `flake8-pyi` and `flake8-bugbear`. As with our other checks, running
+Our code is also linted using `flake8`, with plugins `flake8-pyi`
+`flake8-bugbear`, and flake8-noqa. As with our other checks, running
```
Comment (JelleZijlstra) [inline CONTRIBUTING.md]:
```suggestion
`flake8-bugbear`, and `flake8-noqa`. As with our other checks, running
```

Comment (JelleZijlstra) [reply CONTRIBUTING.md]: Or maybe I was looking at an old version?

Comment (AlexWaygood) [reply CONTRIBUTING.md]: Yeah, I already made this change!

Comment (JelleZijlstra) [reply CONTRIBUTING.md]: Weird that it's still showing "Commit suggestion" as an option

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**